### DISCUSSION TITLE
TimeZone object type is being generated as String

### DISCUSSION THREAD

**Discussion by mmriyadhgit** [Netflix/dgs-framework#1321] [Q&A] [answered]
I have declared a type names TimeZone in my schema.
```
type TimeZone {
        timeZoneId: String!,
        name:  String!
}
```

I'm then using this type in another object type, e.g.,

```
type Store {
        name: String!
       timeZone : TimeZone!
} 
```  

Once the code is generated, the timeZone appears as String!
```
public class Store {
  private String name;
  private String timeZone;
......
}
```

I'm wondering why that would be the case and how I could make it to be an Object type. If I declare the type with some prefix, i.e., `SomeTimeZone`, I gets generated a an object.

I'm using version 4.10.5. Also tried the latest.

```
<dependency>
    <groupId>com.netflix.graphql.dgs</groupId>
    <artifactId>graphql-dgs-platform-dependencies</artifactId>
    <version>4.10.5</version>
    <type>pom</type>
    <scope>import</scope>
</dependency>
```

**Comment by srinivasankavitha** [comment, marked as answer]
This is because we have an implicit type mapping in codegen for this : https://github.com/Netflix/dgs-codegen/blob/master/graphql-dgs-codegen-core/src/main/kotlin/com/netflix/graphql/dgs/codegen/generators/java/TypeUtils.kt#L41

We just merged a PR to fix this, https://github.com/Netflix/dgs-codegen/pull/492, so it will be available in the release on Monday.

**Reply by mmriyadhgit** [reply]
@srinivasankavitha Thanks for fixing this. Has this been released? Would/Could it also be released on the 4.x.x?

**Reply by srinivasankavitha** [reply]
Yes this is released in codegen.
